src/Compilers/CSharp/Test/Emit/PDB/TypeDefinitionDocumentTests.cs

```diff
@@ -0,0 +1,446 @@
+    int y = 1;
+}
+";
+
+            TestTypeDefinitionDocuments(new[] { source });
+        }
+
+        [Fact]
+        public void AbstractMethod()
+        {
+            string source = @"
+abstract class C
+{
+    public abstract void M();
+}
+";
+
+            TestTypeDefinitionDocuments(new[] { source },
+                ("C", "1.cs"));
+        }
+
+        [Fact]
+        public void ExternMethod()
+        {
+            string source = @"
+class C
+{
+    public extern void M();
+}
+";
+
+            TestTypeDefinitionDocuments(new[] { source },
+                ("C", "1.cs"));
+        }
+
+        [Fact]
+        public void Interfaces()
+        {
+            string source1 = @"
+interface I1
+{
+}
+
+partial interface I2
+{
+    public void F();
+}
+";
+            string source2 = @"
+partial interface I2
+{
+}
+";
+            TestTypeDefinitionDocuments(new[] { source1, source2 },
+                ("I1", "1.cs"),
+                ("I2", "1.cs, 2.cs"));
+        }
+
+        [Fact]
+        public void Record()
+        {
+            string source = @"
+record R(int X);
+";
+
+            // The compiler synthesized methods have document info so we don't expect a type document
+            TestTypeDefinitionDocuments(new[] { source, IsExternalInitTypeDefinition },
+                ("IsExternalInit", "2.cs"));
+        }
+
+        [Fact]
+        public void Record_SynthesizedMember()
+        {
+            string source = @"
+record R(int X)
+{
+    protected virtual bool PrintMembers(System.Text.StringBuilder builder)
+    {
+        return true;
+    }
+}
+";
+
+            TestTypeDefinitionDocuments(new[] { source, IsExternalInitTypeDefinition },
+                ("IsExternalInit", "2.cs"));
+        }
+
+        [Fact]
+        public void Enum()
+        {
+            string source = @"
+enum E
+{
+}
+
+enum E2
+{
+    A,
+    B
+}
+";
+
+            TestTypeDefinitionDocuments(new[] { source },
+                ("E", "1.cs"),
+                ("E2", "1.cs"));
+        }
+
+
+        [Fact]
+        public void Delegate()
+        {
+            string source = @"
+delegate void D(int a);
+
+class C
+{
+    void M()
+    {
+        var x = (int a, ref int b) => a;
+    }
+}
+";
+
+            TestTypeDefinitionDocuments(new[] { source },
+                ("D", "1.cs"));
+        }
+
+        [Fact]
+        public void AnonymousTypes()
+        {
+            string source = @"
+class C
+{
+    void M()
+    {
+        var x = new { Goo = 1, Bar = ""Hi"" };
+    }
+}
+";
+
+            TestTypeDefinitionDocuments(new[] { source });
+        }
+
+        [Fact]
+        public void LineDirectives()
+        {
+            string source = @"
+class C
+{
+#line 1 ""C.cs""
+    void M()
+    {
+    }
+#line default
+}
+
+class D
+{
+#line 1 ""D.cs""
+    private int _x = 1;
+    private int X { get; set; } = 1;
+}
+
+#line 1 ""E.cs""
+class E
+{
+    void M()
+    {
+    }
+}
+
+#line 1 ""F.cs""
+class F
+{
+}
```

Comment:
These are interesting cases. When we navigate to E we'd find that M has location "E.cs". For F we'd find "F.cs". But those files actually do not contain the type declaration. They are mapped locations. So that will not help us. 

I think we need to use the original location of the declarations, not the mapped one. In other words, ignore #line directives.